File: sum_all.rb
l = [1, 2, 3, [1, [2, 3]]]

def sum_all list
	if list.length == 0
		0
	elsif list[0].respond_to? 'each'
		sum_all(list[0]) + sum_all(list[1..-1])	
	else	
		list[0] + sum_all(list[1..-1])
	end
end

puts sum_all l

File: sum_all.js
l = [1, 2, 3, [1, [2, 3]]];
var LEN = 6;

function sumAll (list) {
	if (list.length == 0) {
		return 0;
	} else if (list[0].length > 0) {
		return sumAll(list[0]) + sumAll(list.slice(1, LEN));
	} else {
		return list[0] + sumAll(list.slice(1, LEN));
	}
}

console.log(sumAll(l));